clients/drcachesim/tracer/tracer.cpp

```diff
@@ -174,6 +175,24 @@ bbdup_duplication_enabled()
     return attached_midway || bbdup_instr_counting_enabled();
 }
 
+// If we have both BBDUP_MODE_TRACE and BBDUP_MODE_L0_FILTER, then L0 filter is active
+// only when mode is BBDUP_MODE_L0_FILTER
+static void
+get_L0_filters(uintptr_t mode, OUT bool *l0i_enabled, OUT bool *l0d_enabled)
+{
+    if (op_L0_filter_until_instrs.get_value()) {
+        if (mode != BBDUP_MODE_L0_FILTER) {
+            *l0i_enabled = false;
+            *l0d_enabled = false;
+            return;
+        }
+    }
+
+    *l0i_enabled = op_L0I_filter.get_value();
```

Comment:
So L0_filter_until_instrs requires the user to set these too? Document them by the L0_filter_until_instrs flag def.

Reply:
Oh I see we set it automatically.

Reply:
Yes we set them both if none are specified. I will document this.

Reply:
Documented in help message.